File: benchmarks/python/prime_benchmark.py
import time
import random

def count_primes(n):
    is_prime = [True] * (n + 1)
    is_prime[:2] = [False, False]
    for i in range(2, int(n ** 0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, n + 1, i):
                is_prime[j] = False
    return sum(is_prime)

# Generate a random number between 100000000 and 100000100
random_max = random.randint(100000000, 100000100)

# First run (not timed)
count_primes(random_max)

# Second run (timed)
start_time = time.time()
count_primes(random_max)
end_time = time.time()

print(f'{{"type": "prime", "time_ns": {int((end_time - start_time) * 1e9)}}}')

File: benchmarks/csharp/PrimeBenchmark.cs
// File: Program.cs

using System;
using System.Diagnostics;

class PrimeBenchmark {
    static void Main(string[] args) {
        Random random = new Random();
        int randomMax = 10000000 + random.Next(101); // Random number between 10000000 and 10000100

        // First run (not timed)
        CountPrimes(randomMax);

        // Second run (timed)
        var stopwatch = Stopwatch.StartNew();
        CountPrimes(randomMax);
        stopwatch.Stop();

        long time_ns = stopwatch.ElapsedTicks * (1_000_000_000L) / Stopwatch.Frequency;
        Console.WriteLine($"{{\"type\": \"prime\", \"time_ns\": {time_ns}}}");
    }

    private static int CountPrimes(int n) {
        bool[] isPrime = new bool[n + 1];
        for (int i = 2; i <= n; i++) {
            isPrime[i] = true;
        }
        for (int i = 2; i * i <= n; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= n; j += i) {
                    isPrime[j] = false;
                }
            }
        }
        int count = 0;
        for (int i = 2; i <= n; i++) {
            if (isPrime[i]) count++;
        }
        return count;
    }
}